include/deal.II/grid/tria_iterator.h

```diff
@@ -1036,7 +1014,7 @@ template <typename Accessor>
 inline const Accessor *
 TriaRawIterator<Accessor>::operator->() const
 {
-  return &(this->operator*());
+  return &accessor;
```

Comment:
This does not actually improve anything. `operator->` still returns a pointer to a sub-object, and if that is a sub-object to a temporary, you're still screwed. So while
```
  auto face = *cell->face(f);
```
actually works,
```
  auto face = cell->face(f)->operator->();
```
does not. I'm not sure that's such a big improvement.

Reply:
@bangerth But `operator->` is actually accessing the underlying object while the temporary is alive,
 while `operator*` returns a derferenced object. There is a reason these are two different operations.